passes/techmap/extract_reduce.cc

```diff
@@ -154,36 +180,54 @@ struct ExtractReducePass : public Pass
 					cur_supercell.insert(x);
 
 					auto a = sigmap(x->getPort("\\A"));
-					log_assert(a.size() == 1);
-					// Must have only one sink
-					// XXX: Check that it is indeed this node?
-					if (sig_to_sink[a[0]].size() + port_sigs.count(a[0]) == 1)
+					if (a.size() == 1)
```

Comment:
The way the code used to work was that any cells that were in the BFS queue have already been checked to ensure that they are appropriate. Therefore, I don't think check should be added here. The code previously had `log_assert` statements to ensure that cells had the expected number of inputs, but your new code will just fail silently if something unexpected happens.

Reply:
I removed the guard and it will give assert. These lines will add more than single bit cells to BFS queue.
cell_a may be any width.
```
								Cell* cell_a = sig_to_driver[a[0]];
								if (((cell_a->type == "$_AND_" && gt == GateType::And) ||
									 (cell_a->type == "$_OR_" && gt == GateType::Or) ||
									 (cell_a->type == "$_XOR_" && gt == GateType::Xor) ||
									 (cell_a->type == "$and" && gt == GateType::And) ||
									 (cell_a->type == "$or" && gt == GateType::Or) ||
									 (cell_a->type == "$xor" && gt == GateType::Xor)
									 ))
								{
									// The cell here is the correct type, and it's definitely driving only
									// this current cell.
									bfs_queue.push_back(cell_a);

```

Reply:
Please try with this @rqou .
```
read_verilog this.v
prep
extract_reduce
```
```
module and_reduce_get_bit (a, c, d, y);
input [3:0] a, c, d;
output y;
wire [3:0] tmp;
    assign tmp = (c & d);
    assign y = (a[3] & a[2] & a[1] & a[0] & tmp[3]);
endmodule
```